peps/pep-0751.rst

```diff
@@ -0,0 +1,901 @@
+PEP: 751
+Title: A file format to list Python dependencies for installation reproducibility
+Author: Brett Cannon <brett@python.org>
+Status: Draft
+Type: Standards Track
+Topic: Packaging
+Created: 24-Jul-2024
+Replaces: 665
+
+Abstract
+========
+
+This PEP proposes a new file format for specifying what dependencies to
+(possibly) install into a Python environment for consistent installation
+repoducibility. The format is designed to be human-readable but
+machine-generated. Installers consuming the file should not need to resolve
+which listed dependencies to install based on other packages listed in the lock
+file, but instead evaluate each dependency in question in isolation.
+
+
+Motivation
+==========
+
+Currently, there is no standard way to specify what top-level dependencies one
+would like to see installed into a Python environment and then subsequently
+record what was eventually installed in a file (called a *lock file*).
+Considering there are four well-known solutions to this problem in the
+community (``pip freeze``, pip-tools_, Poetry_, and PDM_), there seems to be an
+appetite for lock files in general.
```

Comment:
Perhaps a hedge, but [other](https://docs.astral.sh/uv/projects/#lock-file) [projects](https://pixi.sh/v0.20.0/features/lockfile/) of arguable popularity have formats, and this stops arguments over the cutoff for 'well-known'!
```suggestion
Considering there are at least four well-known solutions to this problem in the
community (``pip freeze``, pip-tools_, Poetry_, and PDM_), there seems to be an
appetite for lock files in general.
```